docs/_newsfragments/2071.newandimproved.rst

```diff
@@ -0,0 +1,6 @@
+Added the :attr:`~.CompiledRouterOptions.allow_on_request` router option that
```

Comment:
I don't particularly like this name `allow_on_request`.
now the hard bit, coming up with a better name. I have come up with:
- `use_on_request`/`enable_on_request`
- `enable_default_responder`
- `on_request_enabled`/`on_request_allowed`

but I'm not sure they are any better

Reply:
I also commented on this, but didn't even come up with anything better then. :grimacing: 

But my top suggestions now, in addition to yours above, are:
* `default_to_on_request`
* `generic_on_request`
* `use_generic_on_request`
* `fallback_to_on_request`

Maybe let's just pick `default_to_on_request` and move on? :smiling_imp:

Reply:
I also like the default_to_on_request.